src/components/Signup.tsx

```diff
@@ -1,51 +1,67 @@
 import React, { useState } from "react";
+import { useRouter } from "next/router";
 import style from "./Signup.module.scss";
 import Button from "./Button";
 import LinkButton from "./LinkButton";
-import TextInput from "./TextInput";
+import CustomInput from "./CustomInput";
+import { registerUser } from "../api/SignUp";
 
 const Signup: React.FC = () => {
   const [firstName, setFirstName] = useState("");
   const [lastName, setLastName] = useState("");
   const [email, setEmail] = useState("");
   const [password, setPassword] = useState("");
 
+  const router = useRouter();
+
   const handleSignup = (e: React.FormEvent) => {
     e.preventDefault();
-    console.log("First name:", firstName);
-    console.log("Last name:", lastName);
-    console.log("E-mail:", email);
-    console.log("Password:", password);
+
+    const requestData = {
+      email,
+      name: `${firstName} ${lastName}`,
+      password,
+    };
+
+    registerUser(requestData).then((response) => {
```

Comment:
Handling promises with .then is completely valid. We gonna use solely async/await and there is no need to mix it up. Please do that here